src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasPointAndLineSlicer.java

```diff
@@ -167,34 +166,26 @@ private boolean allLineTagsEqual(final List<Geometry> slices)
      */
     private List<Geometry> convertToJtsGeometryAndSlice(final Line line)
     {
-        List<Geometry> result;
         final long lineIdentifier = line.getIdentifier();
 
         // Create the JTS Geometry from Line
         Geometry geometry;
-
-        if (isAtlasEdge(line))
-        {
-            geometry = JTS_POLYLINE_CONVERTER.convert(line.asPolyLine());
-        }
-        else if (line.isClosed())
+        if (line.isClosed() && !isAtlasEdge(line))
         {
-            // A Polygon
             geometry = JTS_POLYGON_CONVERTER.convert(new Polygon(line));
         }
         else
         {
-            // A PolyLine
             geometry = JTS_POLYLINE_CONVERTER.convert(line.asPolyLine());
         }
 
         // Slice the JTS Geometry
-        result = sliceGeometry(geometry, line);
+        List<Geometry> result = sliceGeometry(geometry, line);
 
         final boolean multipolygonResult = Iterables.stream(result)
                 .anyMatch(geom -> geom instanceof org.locationtech.jts.geom.Polygon
                         && ((org.locationtech.jts.geom.Polygon) geom).getNumInteriorRing() > 0);
-        if ((result == null || result.isEmpty()) && line.isClosed() || multipolygonResult)
+        if (result.isEmpty() && line.isClosed() || multipolygonResult)
```

Comment:
Just checking that the null-check removal here was intentional.

Reply:
it was! refactor included going ahead and returning an empty collection if slicing was unnecessary, rather than a mix of null or empty